Tools/paletteable sprite editor/js/lib/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+				corfinal = (r + g + b) / 3;
+				r = g = b = corfinal;
+				 
+				r = this.truncate(r + (depth * 2));
+				g = this.truncate(g + depth);
+			
+				//Define a nova cor do ponto...
+				imageOut.setIntColor(x, y, imageIn.getAlphaComponent(x, y), r, g, b);
+			}
+		}
+	}
+
+	/**
+	 * Sets the RGB between 0 and 255
+	 * @param a
+	 * @return
+	 */
+	Sepia.prototype.truncate = function(a) {
+		if      (a <   0) return 0;
+		else if (a > 255) return 255;
+		else              return a;
+	}
+
+	function Thresholding(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+		
+		this.threshold = null;
+		this.thresholdRange = null;
+		this.neighborhood = null;
+		this.range = null;
+	}
+
+	Thresholding.prototype.load = function(){
+		
+		// Attributes
+		this.setAttribute("threshold", 125);
+		this.setAttribute("thresholdRange", -1);
+		this.setAttribute("neighborhood", -1);
+		this.setAttribute("range", -1);
+		
+		this.pluginGray = new GrayScale();
+	}
+	
+	Thresholding.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		this.threshold = this.getAttribute("threshold");
+		this.thresholdRange = this.getAttribute("thresholdRange");
+		this.neighborhood = this.getAttribute("neighborhood");
+		this.range = this.getAttribute("range");
+		
+		if(this.thresholdRange == -1){
+			this.thresholdRange = 255-threshold;
+		}
+		
+		this.pluginGray.process(imageIn, imageOut, attributesOut, mask, previewMode);
+		
+		var bmask = mask.getMask();
+		
+		if(this.neighborhood == -1 && this.range == -1){
+			this.hardThreshold(imageIn, imageOut, bmask);
+		}
+		else{
+			this.contrastThreshold(imageIn, imageOut);
+		}
+				
+	}
+	
+	Thresholding.prototype.hardThreshold = function(imageIn, imageOut, mask){
+		for(var y=0; y<imageIn.getHeight(); y++){
+			for(var x=0; x<imageIn.getWidth(); x++){
+				if(mask != null && !mask[x][y]){
+					continue;
+				}
+				
+				var gray = imageIn.getIntComponent0(x,y); 
+				if(gray < this.threshold || gray > this.threshold+this.thresholdRange){
+					imageOut.setIntColor(x, y, imageIn.getAlphaComponent(x,y), 0,0,0);
+				}
+				else{
+					imageOut.setIntColor(x, y, imageIn.getAlphaComponent(x,y), 255,255,255);
+				}				
+			}
+		}	
+	}
+	
+	Thresholding.prototype.contrastThreshold = function(imageIn, imageOut){
+		this.range = 1;
+		for (var x = 0; x < imageIn.getWidth(); x++) {
+			for (var y = 0; y < imageIn.getHeight(); y++) {
+				if(checkNeighbors(x,y, neighborhood, neighborhood, imageIn)){
+					imageOut.setIntColor(x,y,0,0,0);
+				}
+				else{
+					imageOut.setIntColor(x,y,255,255,255);
+				}
+			}
+		}
+	}
+	
+	Thresholding.prototype.checkNeighbors = function(x, y, neighborhoodX, neighborhoodY, img){
+		
+		var color;
+		var z=0;
+		
+		color = img.getIntComponent0(x, y);
+		
+		for(var i=0-neighborhoodX; i<=neighborhoodX; i++){
+			for(var j=0-neighborhoodY; j<=neighborhoodY; j++){
+				if(i == 0 && j == 0){
```

Comment:
Codacy found an issue: [Use ===/!== to compare with true/false or Numbers](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)